### PR TITLE
[needs-docs] Add else rule UI option in rule based widget

### PR DESCRIPTION
**Pull request by NathanW2**
Finally adds a Else UI item. Making people type ELSE and watching them hit test for it to error was painful. So lets finally fix it :)

image

### CODE REVIEW

File: src/app/qgsrulebasedlabelingwidget.cpp [hunk 1]
```diff
@@ -626,6 +628,8 @@ QgsLabelingRulePropsWidget::QgsLabelingRulePropsWidget( QgsRuleBasedLabeling::Ru
   connect( mScaleRangeWidget, &QgsScaleRangeWidget::rangeChanged, this, &QgsLabelingRulePropsWidget::widgetChanged );
   connect( groupSettings, &QGroupBox::toggled, this, &QgsLabelingRulePropsWidget::widgetChanged );
   connect( mLabelingGui, &QgsTextFormatWidget::widgetChanged, this, &QgsLabelingRulePropsWidget::widgetChanged );
+  connect( mFilterRadio, &QRadioButton::toggled, this, [ = ]( bool toggled ) { filterFrame->setEnabled( toggled ) ; } );
+  connect( mElseRadio, &QRadioButton::toggled, this, [ = ] { editFilter->setText( QStringLiteral( "ELSE" ) );} );
```
Comment (nyalldawson) [inline src/app/qgsrulebasedlabelingwidget.cpp]: I think you want

    connect( mElseRadio, &QRadioButton::toggled, this, [ = ]( bool toggled ) {  if ( togggled ) editFilter->setText( QStringLiteral( "ELSE" ) );} );

File: src/gui/symbology/qgsrulebasedrendererwidget.cpp [hunk 2]
```diff
@@ -798,7 +806,8 @@ void QgsRendererRulePropsWidget::testFilter()
 
 void QgsRendererRulePropsWidget::apply()
 {
-  mRule->setFilterExpression( editFilter->text() );
+  QString filter = mElseRadio->isChecked() ? "ELSE" : editFilter->text();
```
Comment (nyalldawson) [inline src/gui/symbology/qgsrulebasedrendererwidget.cpp]: QStringLiteral

**Review by nyalldawson — COMMENTED** [review]
Woohooo!

**Comment by NathanW2** [comment]
@nyalldawson fixed

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Merge?

**Review by nirvn — APPROVED** [review]

**Pull request merged**